src/LIBSVM.jl

```diff
@@ -363,9 +381,57 @@ function svmpredict{T, U<:Real}(model::SVMModel{T},
     for i = 1:ninstances
         output = ccall(fn, Float64, (Ptr{Void}, Ptr{SVMNode}, Ptr{Float64}),
             model.ptr, nodeptrs[i], pointer(decvalues, nlabels*(i-1)+1))
-        class[i] = model.labels[round(Int,output)]
+        if model.param[1].svm_type==Int32(2) # if one class SVM
+          class[i]=output
+        else
+          class[i] = model.labels[round(Int, output)]
+        end
     end
 
     (class, decvalues)
 end
+
+function svmpredict!{T, U<:Real}(svmpredict_out::Tuple{Array{Any,1},Array{Float64,2},Array{LIBSVM.SVMNode,2},Array{Ptr{LIBSVM.SVMNode},1}}
+                                 , model::SVMModel{T}, instances::AbstractMatrix{U})
+  (class, decvalues, nodes, nodeptrs) = svmpredict_out
+  @assert size(class, 1) == size(decvalues, 2)
+    global verbosity
+    ninstances = size(instances, 2)
+
+    if size(instances, 1) != model.nfeatures
+        error("Model has $(model.nfeatures) but $(size(instances, 1)) provided")
+    end
+
+    instances2nodes!(nodes, nodeptrs, instances)
+    #class = Array(T, ninstances)
+    nlabels = length(model.labels)
+    #decvalues = Array(Float64, nlabels, ninstances)
+
+    verbosity = model.verbose
+    fn = model.param[1].probability == 1 ? svm_predict_probability() :
+        svm_predict_values()
+    for i = 1:ninstances
+        output = ccall(fn, Float64, (Ptr{Void}, Ptr{SVMNode}, Ptr{Float64}),
+            model.ptr, nodeptrs[i], pointer(decvalues, nlabels*(i-1)+1))
+        if model.param[1].svm_type==Int32(2) # if one class SVM
+          class[i]=output
+        else
+          class[i] = model.labels[round(Int, output)]
+        end
+    end
+
+    (class, decvalues)
+end
```

Comment:
Likewise, better to avoid duplicating this code twice and instead define `svmpredict` to call `svmpredict!`.